test/util.py

```diff
@@ -56,6 +64,160 @@ def info_to_usermeta(info: list[int | str | dict[str, str]]) -> dict[str, str]:
     return info[10]
 
 
+def _save_set(
```

Comment:
abstract out functions for creating/saving sets of various types (empty, full, search)